docs/wiki/The library - Module design.md

```diff
@@ -588,83 +588,7 @@ Test file (`deploy.test.bicep`) guidelines:
-    // Parameters //
-    // ========== //
-    @description('Optional. The name of the resource group to deploy for testing purposes')
-    @maxLength(90)
-    param resourceGroupName string = 'ms.analysisservices.servers-${serviceShort}-test-rg'
-
-    @description('Optional. The location to deploy resources to')
-    param location string = deployment().location
-
-    @description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
-    param serviceShort string = 'asscom'
-
-    // =========== //
-    // Dependencies //
-    // =========== //
-
-    // General resources
-    // =================
-    resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
-      name: resourceGroupName
-      location: location
-    }
-
-    module resourceGroupResources 'dependencies.bicep' = {
-      scope: resourceGroup
-      name: '${uniqueString(deployment().name, location)}-nestedDependencies'
-      params: {
-        managedIdentityName: 'dep-<<namePrefix>>-msi-${serviceShort}'
-      }
-    }
-
-    // Diagnostics
-    // ===========
-    module diagnosticDependencies '../../../../.shared/dependencyConstructs/diagnostic.dependencies.bicep' = {
-      scope: resourceGroup
-      name: '${uniqueString(deployment().name, location)}-diagnosticDependencies'
-      params: {
-        storageAccountName: 'dep<<namePrefix>>azsa${serviceShort}01'
-        logAnalyticsWorkspaceName: 'dep-<<namePrefix>>-law-${serviceShort}'
-        eventHubNamespaceEventHubName: 'dep-<<namePrefix>>-evh-${serviceShort}'
-        eventHubNamespaceName: 'dep-<<namePrefix>>-evhns-${serviceShort}'
-        location: location
-      }
-    }
-
-    // ============== //
-    // Test Execution //
-    // ============== //
-
-    module testDeployment '../../deploy.bicep' = {
-      scope: resourceGroup
-      name: '${uniqueString(deployment().name)}-test-${serviceShort}'
-      params: {
-        name: '<<namePrefix>>az${serviceShort}'
-        lock: 'CanNotDelete'
-        skuName: 'S0'
-        roleAssignments: [
-          {
-            roleDefinitionIdOrName: 'Reader'
-            principalIds: [
-              resourceGroupResources.outputs.managedIdentityPrincipalId
-            ]
-            principalType: 'ServicePrincipal'
-          }
-        ]
-        diagnosticLogsRetentionInDays: 7
-        diagnosticStorageAccountId: diagnosticDependencies.outputs.storageAccountResourceId
-        diagnosticWorkspaceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
-        diagnosticEventHubAuthorizationRuleId: diagnosticDependencies.outputs.eventHubAuthorizationRuleId
-        diagnosticEventHubName: diagnosticDependencies.outputs.eventHubNamespaceEventHubName
-      }
-    }
-    ```
+    https://github.com/Azure/ResourceModules/blob/8edc3fb94e43a08f302a14e4bf0cf560f8c39509/modules/Microsoft.AnalysisServices/servers/.test/common/deploy.test.bicep?plain=1#L1-L79
```

Comment:
I wasn't aware this feature/visual exists. Should we removed the surrounding collapsing syntax (i.e., the `<details></details>`)?

Reply:
It depends, that should ideally show the code in place of the link, but I'm not sure about 2 things:

1. if it works from the wiki, being that a different repo
2. how exactly the code is shown. If that falls into a sub-window you can scroll, then we could remove the details.

Let's maybe have it merged to check on the result and then decide if removing the details makes sense. If that works, I'd expect other snippets to be removed and converted to permalinks the same way

Reply:
I agree. Let's check it out. Another thing that would be curious is if it always points to the latest commit (which naturally would be the ideal case).

Reply:
Will approve the PR and then we can go from there